plugins/modules/katello_activation_key.py

```diff
@@ -145,88 +114,151 @@ def override_to_boolnone(override):
-                            activation_key_entity.remove_subscriptions(data={'subscription_id': subscription_id})
+                        ids_to_remove = current_subscription_ids - desired_subscription_ids
+                        if ids_to_remove:
+                            payload = {
+                                'id': activation_key['id'],
+                                'subscriptions': [{'id': item} for item in ids_to_remove],
+                            }
+                            payload.update(scope)
+                            module._resource_action('activation_keys', 'remove_subscriptions', payload)
+                        ids_to_add = desired_subscription_ids - current_subscription_ids
+                        if ids_to_add:
+                            payload = {
+                                'id': activation_key['id'],
+                                'subscriptions': [{'id': item, 'quantity': 1} for item in ids_to_add],
+                            }
+                            payload.update(scope)
+                            module._resource_action('activation_keys', 'add_subscriptions', payload)
                     changed = True
 
-            if 'content_overrides' in entity_dict:
-                content_overrides = entity_dict['content_overrides']
-                product_content = activation_key_entity.product_content()
-                current_content_overrides = set(
-                    (product['content']['label'], product['enabled_content_override'])
+            if content_overrides is not None:
+                _, product_content = module._resource_action('activation_keys', 'product_content', params={'id': activation_key['id']})
+                current_content_overrides = {
+                    product['content']['label']: product['enabled_content_override']
                     for product in product_content['results']
                     if product['enabled_content_override'] is not None
-                )
-                desired_content_overrides = set(
-                    (product['label'], override_to_boolnone(product['override'])) for product in content_overrides
-                )
+                }
+                desired_content_overrides = {
+                    product['label']: override_to_boolnone(product['override']) for product in content_overrides
+                }
+                changed_content_overrides = []
+
+                for label, override in desired_content_overrides.items():
+                    if override != current_content_overrides.pop(label, None):
+                        changed_content_overrides.append({'content_label': label, 'value': override})
+                for label in current_content_overrides.keys():
+                    changed_content_overrides.append({'content_label': label, 'reset': True})
+
+                if changed_content_overrides:
+                    if not module.check_mode:
```

Comment:
`_resource_action` already checks for check mode, so no need to repeat that here:

https://github.com/theforeman/foreman-ansible-modules/blob/ec9963682435244742e90f9fe26aaf03a5f4ed3f/plugins/module_utils/foreman_helper.py#L453-L462

Reply:
You are right, i missed that.